vendors/espressif/boards/esp32/CMakeLists.txt

```diff
@@ -173,21 +173,53 @@ target_include_directories(
         "${esp_idf_dir}/components/esp_netif/include"
         "${esp_idf_dir}/components/esp_eth/include"
         "${esp_idf_dir}/components/soc/soc/include"
-        "${AFR_ROOT_DIR}/libraries/coreHTTP/source/dependency/3rdparty/http_parser"
 )
 endif()
 
 target_sources(
     AFR::wifi::mcu_port
     INTERFACE 
         "${afr_ports_dir}/wifi/iot_wifi.c"
-        "${afr_ports_dir}/wifi/iot_softap_wifi_provisioning.c"
-        # Due to nghttp's dependency on it, http_parser is added to generate metadata
-        # so that it is present in the code downloaded from the FreeRTOS console.
-        "${AFR_ROOT_DIR}/libraries/coreHTTP/source/dependency/3rdparty/http_parser/http_parser.c"
-        "${AFR_ROOT_DIR}/libraries/coreHTTP/source/dependency/3rdparty/http_parser/http_parser.h"
 )
 
+# SoftAP Provisioning - Available ONLY when using LWIP stack
+if( NOT AFR_ESP_FREERTOS_TCP )
+    target_include_directories(
+        AFR::wifi::mcu_port
+        INTERFACE
+            "${AFR_ROOT_DIR}/libraries/coreHTTP/source/dependency/3rdparty/http_parser"
+            "${AFR_ROOT_DIR}/libraries/3rdparty/mbedtls/include"
+            "${esp_idf_dir}/components/protobuf-c/protobuf-c"
+            "${esp_idf_dir}/components/protocomm/proto-c"
+            "${esp_idf_dir}/components/protocomm/src/common"
+            "${esp_idf_dir}/components/wifi_provisioning/proto-c"
+            "${esp_idf_dir}/components/esp_http_server/include"
+    )
+    target_sources(
+        AFR::wifi::mcu_port
+        INTERFACE
+            "${afr_ports_dir}/wifi/iot_softap_wifi_provisioning.c"
+            "${AFR_ROOT_DIR}/libraries/coreHTTP/source/dependency/3rdparty/http_parser/http_parser.c"
+            "${AFR_ROOT_DIR}/libraries/coreHTTP/source/dependency/3rdparty/http_parser/http_parser.h"
+            "${esp_idf_dir}/components/protobuf-c/protobuf-c/protobuf-c/protobuf-c.c"
```

Comment:
There is no need to add individual source files to target sources (these files are added to build list by ESP-IDF build system). You just need to link required ESP-IDF component by adding `idf::<component_name>` in target_link_libraries.

Reply:
Hi @shubhamkulkarni97 

I prefer this, but there are caveats.

Doing something similar in the past, lead to some _very_ nasty bugs where the `http_parser` objects were being used with inconsistent offsets across modules, corrupting the whole thing http parser and eventually the stack.... etc. 

Our `coreHTTP` and `nghttp` offering _slightly_ different versions of `http_parser.o`. Hence the explicit use of http_parser.o. Initially thought `coreHTTP` was better, but for the provisioning `esp_http_server` it shouldn't matter.

I want to implement this mostly-aesthetic change, but would like to defer until I can give it the careful attention it needs. To not block fixes, would prefer to address this is follow-up PR

Reply:
Hi @dachalco, it is fine to add individual source files to avoid issues for now.
Can you please try alternate solution without adding individual sources in follow-up PR? (this will be helpful while upgrading ESP-IDF version in future)

Reply:
Of course, I added it to my current queues so it should be getting addressed in 1-3 weeks